### ISSUE TITLE
A little confused about the code

### ISSUE THREAD

**Issue by 578578wu** [ximinng/SVGDreamer#26]
Hello, thank you for your great work. When I carefully reviewed your code, I found out why the target was randomly generated noise when the mse loss was calculated in the function "train_phi_model". What is the purpose of doing this? I am puzzled. I hope the author can answer it. Thank you very much!
![{AA6266FE-318A-4795-A48C-49AD26509DA5}](https://github.com/user-attachments/assets/866aeca8-58af-4290-9d5f-0b2cb13be8a7)
![{37B892EF-1C83-4A18-8201-8AF1E5C6CEEB}](https://github.com/user-attachments/assets/9866d223-a658-4b92-8ab1-e3dedc12a031)

**Comment by 578578wu** [comment]
Sorry,it seems that the image is not uploaded,the key code is:
 noise = torch.randn_like(latents_phi)
 noisy_latents = self.scheduler.add_noise(latents_phi, noise, t)
 return F.mse_loss(noise_pred, target, reduction="mean")

**Comment by ximinng** [comment]
\phi_model is implemented by SD with lora, and its objective function is the same as SD. Recover samples from random Gaussian noise.

Also, did you review the results of the code? Any other bugs?

**Comment by 578578wu** [comment]
Thank you so much for your response! I believe I understand  now!

Previously, I was stuck in the mindset that during training, the loss should be computed as the MSE between the noise predicted at step t and the "real noise" added at step t. Additionally, I mistakenly thought the "real noise" was scaled_noise.

However, the target predicted by the UNet is actually the original noise, not the scaled_noise.

This is the difference between the two approaches:
```
# Approach 1: Predict the original noise
noise = torch.randn_like(x_0)
loss = F.mse_loss(noise, predicted_noise)

# Approach 2: Predict the scaled noise
scaled_noise = sqrt(1-alpha_bar_t) * noise
loss = F.mse_loss(scaled_noise, predicted_noise)
```

Have fully reproduced and read your code and found no other issues ~

That said, regarding issue25, you mentioned: "In fact, it is simple to change the model_id all to sdxl in config."

This doesn’t seem to work on my end because the pipelines for SD1.5 and SDXL are quite different. Specifically, the encode_prompt function needs to be modified to handle the two text_encoder, and in the "variational_score_distillation" function, the Unet’s input requires additional preparation:
`added_cond_kwargs = {"text_embeds": add_text_embeds, "time_ids": add_time_ids}`

However, the effect of sdxl in my place is not as good as sd1.5, I guess it is also necessary to adjust some hyperparameters?

Your code is not only well-structured and clear but also logically rigorous, which has been incredibly helpful for me as a beginner. 
I hope you can achieve more outstanding research results in the field of computer vision in the future!
![0649CE4A](https://github.com/user-attachments/assets/a9001934-6f82-4e1b-a47b-4873e8bc2ee0)
![0649DA6F](https://github.com/user-attachments/assets/f4176a67-d584-480e-aa70-831365e91943)

**Comment by ximinng** [comment]
First of all, thank you for your interest in our SVGDreamer.

I have tried to change the base model of SVGDreamer from sd21b to sdxl.
Besides changing config, u also need to modify the corresponding diffuser pipeline in the pipeline. 
In addition, I found that the text encoder in sdxl was not the key to affecting the results. Its VAE is a problem to be solved.

I didn't open source the svgdreamer sdxl version, but here's a reference: https://github.com/ximinng/VectorFusion-pytorch/blob/master/methods/painter/vectorfusion/LSDS_SDXL_pipeline.py

**Comment by 578578wu** [comment]
Thank you for your reply！

this reference file is very helpful to me~
